src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4356,7 +4361,11 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             {
                 if (HasImplicitTypeArguments(node))
                 {
-                    var binder = (node as BoundCall)?.BinderOpt ?? (node as BoundCollectionElementInitializer)?.BinderOpt ?? throw ExceptionUtilities.UnexpectedValue(node);
+                    var binder =
+                        (node as BoundCall)?.BinderOpt
```

Comment:
Lets refactor this into a switch expression. Will be much clearer.

Reply:
The switch looks like this:

```csharp
                    var binder = node switch
                    {
                        BoundCall { BinderOpt: { } x } => x,
                        BoundCollectionElementInitializer { BinderOpt: { } x } => x,
                        BoundForEachStatement { EnumeratorInfoOpt: { Binder: var x } } => x,
                        _ => throw ExceptionUtilities.UnexpectedValue(node)
                    };
```

I'm not sure that's any clearer.
The issue is that it has to match the type, and the binder could be null as well.

Reply:
Unless we're happy to NullReferenceException instead of InvalidOperationException, seeing as in practice they never should be null. It'll upset NRT checking once that's enabled though so I don't think it's a good idea.

Reply:
Personally, I find this better to read, just need to change that `var` pattern to a `{ }` and it's equivalent code.